### DISCUSSION TITLE
Using `Literal` with `@overload`: autocomplete behavior

### DISCUSSION THREAD

**Discussion by Viicos** [microsoft/pyright#6732] [Q&A] [answered]
Consider the following example:

```python
from typing import Literal, overload


class A:
    @overload
    def __init__(self, type_: Literal["int", "INT"], boolean: Literal[True]) -> None:
        ...

    @overload
    def __init__(self, type_: Literal["int", "INT"], boolean: Literal[False] = ...) -> None:
        ...

    @overload
    def __init__(self, type_: Literal["str", "STR"], boolean: Literal[True]) -> None:
        ...

    @overload
    def __init__(self, type_: Literal["str", "STR"], boolean: Literal[False] = ...) -> None:
        ...

A("...")
```

[playground](https://pyright-play.net/?pythonVersion=3.13&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAySMApiAIYA2ANFGAG5mVjkAmAUJwMaXkDOfKAEEAXOygSoAAQZMWHSVFYlgUAPprUxDQAo%2BJSsFrwEJNSMLEyVANoAiVDDu07ASQByAFTsBdWgCMwMEoLIlIKShtPEABXEh8ASigAWgA%2BKHcwFBIxRUUAOkLORRlGEGY2cUllVQ0tGF19Q2NEM1CrCPtHZyg3L18AoJDLcNsAMSp9HygAXihC-KS0jKycqrz5ouLJUrlKxRr1TRRtNT0DI1hW8xHrSLs%2BGBAeuwBlTwAlAahA4PbRyLROKJFLpTLZXIbTb5bYSXbleTrJQqI71RoXFqmG5hO72R7PFzvL5%2BH5Df64iaUKazaFLMGrSEbBacIQ6Ox2BJAA)

I've simplified the example so doesn't really mean anything currently, but for the context I'm writing a stub file where `A` is generic, and I'm type hinting `self` depending on the `type_` value.

Currently, the shown auto completion is:
![image](https://github.com/microsoft/pyright/assets/65306057/a978d3b3-c3a0-42e2-bb60-5e22a54904c7)

Is there a way to have the auto completion show all literals? i.e. `str, STR, int, INT`.

Thanks!

**Comment by erictraut** [comment, marked as answer]
This does look like a potential bug. The completion provider (the logic that provides completion suggestions) in pyright is maintained by the pylance team. I recommend opening an issue in the [pylance-release](https://github.com/microsoft/pylance-release) issue tracker.

**Reply by Viicos** [reply]
Thanks @erictraut, the line between pyright and pylance is hard to get sometimes, I'll open an issue there.

**Reply by Viicos** [reply]
Turns out it is already tracked here: https://github.com/microsoft/pylance-release/issues/4428
